kola/tests/rpmostree/status.go

```diff
@@ -36,7 +36,7 @@ func init() {
 
 var (
 	// Regex to extract version number from "rpm-ostree status"
-	rpmOstreeVersionRegex string = `^Version: (\d+\.\d+\.?\d*).*`
+	rpmOstreeVersionRegex string = `^Version: (\d+.*) \(.*`
```

Comment:
We should probably specifically call out only the allowed values.

Maybe something like:
`^Version: ([\d\.]+).*`

Reply:
Works as expected as well:

```
>>> re.findall("^Version: ([\d\.]+).*", 'Version: 400.7.20190227.0 (2019-02-27T16:19:34Z)')
['400.7.20190227.0']
```

Reply:
I tried a few more version strings we've seen in the `ostree` world and they all seem to work, as long as they are just digits.  If we introduce a letter, the results are not expected.  I think this is fine, as I don't expect us to use any letters in version strings.

```
>>> v = re.compile("^Version: ([\d.]+).")
>>> s = "Version: 400.7.20190227.0 (2019-02-27T16:19:34Z)"
>>> v.findall(s)
['400.7.20190227.0']
>>> s = "Version: 47.334 (2019-02-27T16:19:34Z)"
>>> v.findall(s)
['47.334']
>>> s = "Version: 29.20190226.0 (2019-02-27T16:19:34Z)"
>>> v.findall(s)
['29.20190226.0']
>>> s = "Version: 29.20190226.a (2019-02-27T16:19:34Z)"
>>> v.findall(s)
['29.20190226.']
```

Reply:
Hmm, it seems that the POSIX brackets are removing the need to escape the `.`; I would have expected `[\d.]` to be essentially a useless character class as it includes the wildcard but it's actually matching `[0-9]` or the `.` character.

Reply:
Heh, I saw this comment after it was merged.  I changed the regex to include the escaped `.`, ala
`"^Version: ([\d\.]+).*"` and things worked the same for the test strings:

```
>>> v = re.compile("^Version: ([\d\.]+).*")
>>> s = "Version: 400.7.20190227.0 (2019-02-27T16:19:34Z)"
>>> v.findall(s)
['400.7.20190227.0']
>>> s = "Version: 47.334 (2019-02-27T16:19:34Z)"
>>> v.findall(s)
['47.334']
>>> s = "Version: 29.20190226.0 (2019-02-27T16:19:34Z)"
>>> v.findall(s)
['29.20190226.0']
>>> s = "Version: 29.20190226.a (2019-02-27T16:19:34Z)"
>>> v.findall(s)
['29.20190226.']
```

So good, either way.

Reply:
Yeah, the non-escaped `.` inside of a character class definition seems to be the same as an escaped `.`, not a wild-card. TIL